### PR TITLE
Store module prefills under Twinkle.prefill

### PR DESCRIPTION
**Pull request by nardog**
First stab at #1682. Gets relevant properties from the URL query on initialization and stores them under `Twinkle.prefill`, replacing mw.util.getParamValue(), so that the prefills may be configured after page load.

### CODE REVIEW

**Comment by NovemLinguae** [comment]
No need to squash PRs on your end. We will squash when we accept the PR. There's a button that says "squash" that makes it easy.

Linter is currently failing. The message is below. If you don't mind, please fix when you get a chance.

```
787:1  error  Expected indentation of 3 tabs but found 2  indent

1 problem (1 error, 0 warnings)
1 error and 0 warnings potentially fixable with the `--fix` option.

**Comment by nardog** [comment]
Thanks, I saw the error but somehow I convinced myself it was a false positive. Fixed.

**Comment by NovemLinguae** [comment]
Is this purely a refactor that doesn't change any behavior?

Is your goal to eventually prefill some stuff when hitting the [rollback] [vandalism] buttons on a user contributions page?

image

**Comment by nardog** [comment]
> Is this purely a refactor that doesn't change any behavior?

No, the patch should be a complete solution for #1682.

> Right now hitting those buttons doesn't visit the user talk page at all. It redirects to the article.

It's effective only if you have "Don't reload the page when rolling back from contributions or recent changes" on. Then if you roll back on a contributions page and open ARV, Warn, or Wel *on the same page*, certain fields in the dialog should be prefilled based on the last performed rollback.

**Review by NovemLinguae — CHANGES_REQUESTED** [review]

File: twinkle.js [hunk 1]
```diff
@@ -508,6 +508,20 @@ Twinkle.summaryAd = ' ([[WP:TW|TW]])';
 // ensure MOS:ORDER
 Twinkle.hatnoteRegex = 'short description|hatnote|main|correct title|dablink|distinguish|for|further|selfref|year dab|similar names|highway detail hatnote|broader|about(?:-distinguish| other people)?|other\\s?(?:hurricane(?: use)?s|people|persons|places|ships|uses(?: of)?)|redirect(?:-(?:distinguish|synonym|multi))?|see\\s?(?:wiktionary|also(?: if exists)?)';
 
+// Prefills in ARV/Warn/Welcome set by fluff
```
[the remark this answers is not in the document]
Comment (nardog) [reply twinkle.js:511]: Is this good? The primary use case of the prefills is when a user talk is opened after a rollback or from the "talk" link on the right side of a diff.

(I think we should eventually stop using the URL search string for prefills and switch to [postMessage()](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) or something instead, which this patch makes easier.)

**Comment by NovemLinguae** [comment]
Took me quite awhile to figure out what this code is doing. I finally stopped looking at diffs and figured out that Twinkle peppers in some [rollback] and [vandalism] tags into the user's contribs. Can start testing properly now.

image

**Pull request merged**